packages/local_auth/local_auth_android/android/src/main/java/io/flutter/plugins/localauth/AuthResultErrorCodes.java

```diff
@@ -0,0 +1,34 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+package io.flutter.plugins.localauth;
+
+/** Exception codes for `PlatformException` in Flutter returned by `authenticate`. */
+class AuthResultErrorCodes {
+  /**
+   * Indicates that the user has not yet configured a passcode (iOS) or PIN/pattern/password
+   * (Android) on the device.
+   */
+  static final String PASSCODE_NOT_SET = "PasscodeNotSet";
+
+  /** Indicates the user has not enrolled any biometrics on the device. */
+  static final String NOT_ENROLLED = "NotEnrolled";
+
+  /** Indicates the device does not have hardware support for biometrics. */
+  static final String NOT_AVAILABLE = "NotAvailable";
+
+  /** Indicates the device operating system is unsupported. */
+  static final String OTHER_OPERATING_SYSTEM = "OtherOperatingSystem";
+
+  /** Indicates the API is temporarily locked out due to too many attempts. */
+  static final String LOCKED_OUT = "LockedOut";
+
+  /**
+   * Indicates the API is locked out more persistently than [lockedOut]. Strong authentication like
+   * PIN/Pattern/Password is required to unlock.
+   */
+  static final String PERMANENTLY_LOCKED_OUT = "PermanentlyLockedOut";
+
+  /** Indicates that the biometricOnly parameter can"t be true on Windows */
+  static final String BIOMETRIC_ONLY_NOT_SUPPORTED = "biometricOnlyNotSupported";
```

Comment:
Agreed, we should only define strings we are actually using, and referencing the Dart file that these need to be synchronized with in a comment is definitely helpful.

We don't currently have a way to avoid the redundancy. (There's a proposal for constants in Pigeon that would let us address that problem.)

Reply:
We still have some error codes in Android plugin which are not contained in error_codes.dart such as "NotSupported", "NoHardware" and others. Should we add all of them to AuthResultErrorCodes.java or just keep here values matching those in error_codes.dart?

Reply:
@camsim99 @stuartmorgan Any thoughts?

Reply:
I think including them is fine. Maybe just note that those are Android only or Android-specific?